src/test/java/ReaderWriterTest.java

```diff
@@ -0,0 +1,125 @@
+public final class ReaderWriterTest {
+
+  @Test
+  public void validateDatabaseName() throws IOException {
+    System.out.println("validateDatabaseName()");
+    Database database = new Database("validateDatabaseName");
+    /* Validate that place where data is stored is named test. */
+    System.out.println("test = " + database.getDatabaseName());
+    assertEquals("validateDatabaseName", database.getDatabaseName());
+
+    /* Delete directory. */
+    database.getDatabase().delete();
+  }
+
+  @Test
+  public void validateWriterRunId() throws IOException {
+    System.out.println("validateWriterRunId()");
+    Database database = new Database("validateWriterRunId");
+    /* Validate that writer's runId is test. */
+    DbWriter writer = new DbWriter(database, "2ndTestRunId", "writingIn");
+    assertEquals("2ndTestRunId", writer.getRunId());
+
+    /* Delete directory. */
+    database.getDatabase().delete();
+  }
+
+  @Test
+  public void validateReaderWriterDatabaseConnect() throws IOException {
+    System.out.println("validateReaderWriterDatabaseConnect()");
+
+    /*Validate that what writer wrote to database is what reader reads. */
+    /* Initialize Database, reader, and writer. */
+    Database database = new Database("validateConnect");
+    DbWriter writer = new DbWriter(database, "3rdTestRunId", "writingIn");
+    DbReader reader = new DbReader(database, "3rdTestRunId", "writingIn");
+
+    String writeTest = "This is a test string";
+
+    /* Write writeTest into database. */
+    try (OutputStream output1 = writer.write()) {
+      byte[] b = writeTest.getBytes();
+      output1.write(b);
+    }
+
+    String readTest = null;
+
+    /* Read contents of file written to database. */
+    try (Reader reads =
+        new BufferedReader(
+            new InputStreamReader(reader.read(), Charset.forName(StandardCharsets.UTF_8.name())))) {
+      /* Convert inputStream to Stringbuilder. */
+      StringBuilder textBuilder = new StringBuilder();
+      int c = 0;
+      while ((c = reads.read()) != -1) textBuilder.append((char) c);
+      /* Convert Stringbuilder to a string. */
+      readTest = textBuilder.toString();
+    }
+
+    assertEquals(writeTest, readTest);
+
+    /* Delete directory  recursively. */
+    File[] allContents = database.getDatabase().listFiles();
+    if (allContents != null) {
+      for (File file : allContents) {
+        file.delete();
+      }
+    }
+    database.getDatabase().delete();
+  }
+
+  @Test
+  public void validateDatabaseQuery() throws IOException {
+    Database database = new Database("validateDatabaseQuery");
+    ArrayList<String> expectedFiles = new ArrayList<String>();
+
+    /* Add files to Database. */
+    int numOfFiles = 3;
+    String tempRunId;
+    String tempType;
+    for (int i = 0; i < numOfFiles; i++) {
+      tempRunId = "FileNo." + i;
+      tempType = "test";
+      DbWriter writer1 = new DbWriter(database, tempRunId, tempType);
+      writer1.write();
```

Comment:
What's the point of `writer1.write()` when you discard the return value?

Reply:
during write() a file is created with the filename corresponding to the runid and type. this test is only testing that the files are all there, so it doesnt need to know about the contents of them.